src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1230,6 +1230,19 @@ object LazyList extends SeqFactory[LazyList] {
     at(0)
   }
 
+  override def range[A: Integral](start: A, end: A): LazyList[A] =
+    newLL(rangeImpl(start, end, step = Integral[A].one))
+
+  override def range[A](start: A, end: A, step: A)(implicit ev: Integral[A]): LazyList[A] =
+    newLL {
+      if (ev.equiv(step, ev.zero)) throw new IllegalArgumentException("step cannot be 0.")
+      else rangeImpl(start, end, step)
+    }
+
+  private[this] def rangeImpl[A](start: A, end: A, step: A)(implicit ev: Integral[A]): State[A] =
+    if (ev.lt(start, end)) sCons(start, newLL(rangeImpl(ev.plus(start, step), end, step)))
```

Comment:
Uhm, wait, this does not allow reversed ranges, does it? Like `LazyList.range(start = 100, end = 0, step = -2)`
I guess rather than check for `lt` we have to check for `!ev.equiv(start, end)`?

Reply:
Also, this made me think. This could be a good way to test this without a CPU trap. We just request the reversed range and then we can indeed ask for the `head` and check it is bigger than `Int.MaxValue`

Reply:
good point, the check is incorrect, hmm. probably need to check the sign or something as well. I'll have another look shortly

Reply:
my thought is to check the sign of `step` in the 3 parameter overload of `range`, and then pass it as a `Boolean` parameter (probably named `positiveStep`) to `rangeImpl`, thus checking the condition only once rather than for each element

Reply:
I just pushed an implementation that allows for reverse `ranges`.